workbench/test/selenium_test.py

```diff
@@ -19,3 +23,15 @@ def setUp(self):
         # Clear the in-memory key value store, the usage store, and whatever
         # else needs to be cleared and re-initialized.
         reset_global_state()
+
+    @contextmanager
+    def wait_for_page_load(self, timeout=30):
+        """
+        Uses Selenium's built-in "staleness" hook to wait until the page has
+        loaded. For use when clicking a link to ensure that the new page
+        has loaded before selecting elements.
+        """
+        old_page = self.browser.find_element_by_tag_name('html')
+        yield WebDriverWait(self.browser, timeout).until(
+            staleness_of(old_page)
+        )
```

Comment:
What's the value of making this a context manager, rather than a plain function.  It doesn't seem to establish a context, but just mark a timepoint.  Or if it is a context manager, shouldn't it be more like
    
    def wait_for_page_load(timeout=30):
        try:
            yield self.browser.find_element_by_tag_name('html')
        finally:
            WebDriverWait(self.browser, timeout).until(staleness_of(old_page))

And then use it earlier:
 
    with self.wait_for_page_load():  # as old_page:
        link.click()

The way it's used now the new page could possibly load before we even start waiting, and then it will wait until *that* page is stale.